client/my-sites/stats/summary/index.jsx

```diff
@@ -379,13 +385,17 @@ class StatsSummary extends Component {
 							} }
 							rightSection={
 								<div className="stats-module__header-nav-button">
-									<DownloadCsv
-										statType={ statType }
-										query={ moduleQuery }
-										path={ path }
-										period={ this.props.period }
-										skipQuery
-									/>
+									{ shouldGateStatsCsvDownload ? (
+										<DownloadCsvUpsell siteId={ siteId } borderless />
```

Comment:
This doesn't seem to work for Jetpack sites in my testing... but not a blocker atm